### PR TITLE
feat(menu): Add BEM classes

### PR DESCRIPTION
**Pull request by pixelastic**
You know the drill.

- Updated BEM classes, including root, header, footer
- Updated docs

BREAKING CHANGE: The default template now has the count element inside
the link, not outside.

### CODE REVIEW

File: widgets/menu/menu.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ function menu({
   var containerNode = utils.getContainerNode(container);
   var usage = 'Usage: menu({container, facetName, [sortBy, limit, cssClasses.{root,list,item}, templates.{header,item,footer}, transformData, hideWhenResults]})';
 
-  if (container === null || facetName === null) {
+  if (!container || !facetName) {
```
Comment (bobylito) [inline widgets/menu/menu.js]: Why? :) Was there an edge case that we didn't support in this case?

Comment (pixelastic) [inline widgets/menu/menu.js]: Nope, it's a line that had been forgotten in a previous refactoring I guess.

We used to pass default parameters as `container = null, facetName = null` and we then decided to simply pass `container, facetName` with no value and test if they were falsy.

Here we had done only half the job so I changed the second part :)

Comment (bobylito) [inline widgets/menu/menu.js]: Oh we decided that? Cool then. :)

**Comment by bobylito** [comment]
LGTM! :+1:

File: widgets/menu/menu.js (hunk 1 above)
Comment (pixelastic) [inline widgets/menu/menu.js]: It was more an oral decision that anything written down actually ;)

Comment (bobylito) [inline widgets/menu/menu.js]: AH :)

Comment (pixelastic) [inline widgets/menu/menu.js]: Yeah I know, we'll have to have a place to write those styleguide/standards somewhere soon. Just a wiki page would do.

**Pull request merged**